packages/test/test-gc-sweep-tests/src/testDataObjects.ts

```diff
@@ -25,3 +52,153 @@ export class TestDataObject extends DataObject {
+        return channelIds;
+    }
+
+    public getHandleChannelIds(): string[] {
+        return Array.from(this.handleManager.handleOpManagers.keys());
+    }
+
+    private addChannelType(channel: IChannel) {
+        this.root.set(channel.constructor.name, channel.handle);
+        this.root.set(channel.id, channel.handle);
+
+        const handles: IFluidHandle[] = this.root.get(channelHandlesKey) ?? [];
+        handles.push(channel.handle);
+        this.root.set(channelHandlesKey, handles);
+
+        const ids: string[] = this.root.get(channelIdsKey) ?? [];
+        ids.push(channel.id);
+        this.root.set(channelIdsKey, ids);
+    }
+
+    protected async initializingFirstTime(props?: any): Promise<void> {
+        this.addChannelType(ConsensusQueue.create(this.runtime));
+        this.addChannelType(ConsensusRegisterCollection.create(this.runtime));
+        this.addChannelType(Ink.create(this.runtime));
+        this.addChannelType(SharedCell.create(this.runtime));
+        this.addChannelType(SharedCounter.create(this.runtime));
+        // this.addChannelType(SharedDirectory.create(this.runtime));
+        this.addChannelType(SharedMap.create(this.runtime));
+        // this.addChannelType(SharedMatrix.create(this.runtime));
+        // this.addChannelType(SharedString.create(this.runtime));
+    }
+
+    private async assertedGet<T>(key: string): Promise<T> {
+        const handle = this.root.get<IFluidHandle<T>>(key);
+        assert(handle !== undefined, `Expected ${key}!`);
+        const channel = await handle.get();
+        assert(channel !== undefined, `Expected channel to be defined for ${key}!`);
+        return channel;
+    }
+
+    protected async hasInitialized(): Promise<void> {
+        const consensusQueueHandler = new ConsensusQueueHandler(
+            await this.assertedGet<ConsensusQueue>("ConsensusQueue"),
+        );
+        this.handleManager.add(consensusQueueHandler);
+
+        const consensusRegisterCollectionHandler = new ConsensusRegisterCollectionHandler(
+            this.root,
+            await this.assertedGet<ConsensusRegisterCollection<IFluidHandle | null>>("ConsensusRegisterCollection"),
+        );
+        this.handleManager.add(consensusRegisterCollectionHandler);
+
+        const inkHandler = new InkHandler(await this.assertedGet<Ink>("Ink"));
+        this.randomOpHandlers.push(inkHandler);
+
+        const sharedCellHandler = new SharedCellHandler(await this.assertedGet<SharedCell>("SharedCell"));
+        this.handleManager.add(sharedCellHandler);
+
+        const sharedCounterHandler = new SharedCounterHandler(await this.assertedGet<SharedCounter>("SharedCounter"));
+        this.randomOpHandlers.push(sharedCounterHandler);
+
+        const sharedMapHandler = new SharedMapHandler(await this.assertedGet<SharedMap>("SharedMap"));
+        this.handleManager.add(sharedMapHandler);
+        this.randomOpHandlers.push(sharedMapHandler);
+    }
+}
+
+export class HandleManager {
```

Comment:
Should this get its own file?

Reply:
I guess it's pretty small so many not.  Does it need to be exported?

Reply:
We may want to move it back up into the `TestDataObject` class, I had thought there might be more here, but for now, we'll leave it here.